### PR TITLE
cleanup wallet sync logic

### PR DESCRIPTION
**Pull request by joschisan**
The consensus logic is not actually changed by this pr, as None and Some(0) returned by consensus_block_count do not trigger a sync, so we might as well just replace any None with a 0 to begin with.

### CODE REVIEW

File: modules/fedimint-wallet-server/src/lib.rs [hunk 1]
```diff
@@ -986,7 +981,7 @@ impl Wallet {
             "New consensus height, syncing up",
         );
 
-        for height in (old_height + 1)..=(new_height) {
```
Comment (dpc) [inline modules/fedimint-wallet-server/src/lib.rs:989]: Bleh. :D

File: modules/fedimint-wallet-server/src/lib.rs [hunk 2]
```diff
@@ -986,7 +981,7 @@ impl Wallet {
...
+        for height in old_height..new_height {
```
Comment (dpc) [inline modules/fedimint-wallet-server/src/lib.rs:984]: :+1:

**Review by dpc — APPROVED** [review]
I'm a bit scared of consensus refactoring, but seems OK to me  and existing version is really meh.

File: modules/fedimint-wallet-server/src/lib.rs (hunk 1 above)
Comment (dpc) [reply modules/fedimint-wallet-server/src/lib.rs:989]: I think initially this code was operating on "block count" like `bitcoind`s `getblockcount` rpc, which is actually not a block count, but the height of the highest block (so inclusive range), which is probably the dumbest part of `bitcoind`, right before reversting txids.

Then I went on mission to change everything to use real `block_count`, but this code might have got transformed mechanically to match previous behavior. Wouldn't surprise me if I did this it. :D

File: modules/fedimint-wallet-server/src/lib.rs (hunk 2 above)
Comment (dpc) [reply modules/fedimint-wallet-server/src/lib.rs:984]: Though actually... I would rename `s/height/count` everywhere here.

**Review by elsirion — APPROVED** [review]
Agree with @dpc's height->count rename suggestion, but structurally the PR is already a big improvement.

**Pull request merged**